hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/headers/Http1RequestHeadersProvider.java

```diff
@@ -0,0 +1,45 @@
+package pl.allegro.tech.hermes.consumers.consumer.sender.http.headers;
+
+import com.google.common.collect.ImmutableMap;
+import org.eclipse.jetty.http.HttpHeader;
+import pl.allegro.tech.hermes.api.ContentType;
+import pl.allegro.tech.hermes.consumers.consumer.Message;
+import pl.allegro.tech.hermes.consumers.consumer.sender.http.auth.HttpAuthorizationProvider;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static pl.allegro.tech.hermes.api.AvroMediaType.AVRO_BINARY;
+import static pl.allegro.tech.hermes.api.ContentType.AVRO;
+
+public final class Http1RequestHeadersProvider implements HttpRequestHeadersProvider {
+
+    private static final Function<ContentType, String> contentTypeToMediaType = contentType ->
+            AVRO.equals(contentType) ? AVRO_BINARY : APPLICATION_JSON;
+
+    private final Optional<HttpAuthorizationProvider> authorizationProvider;
+
+    public Http1RequestHeadersProvider(Optional<HttpAuthorizationProvider> authorizationProvider) {
```

Comment:
Optionals do not fit well as arguments... I know there may be some places we do it like this, but it's an antipattern.
Let's explicitly keep a field that may be nullable, and just create two constructors.

In `getHeaders(...)` we'll have a simple null-check.

Reply:
I agree. I refactored it to not-optional version on provider level, but `HttpAuthorizationProviderFactory` still operates on `Optional`. Do you think we should change it too?

Reply:
Is it? I can see it returns an `Optional` which is absolutely fine. :)